File: src/main/java/com/kevin/data/model/Counter.java
package com.kevin.data.model;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;
import java.sql.Timestamp;

@Data
@Entity
@Table(name = "counter")
public class Counter implements Serializable {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Integer id;

  @Column(nullable = false)
  private Integer count;

  @Column(nullable = false)
  private Timestamp createdAt = new Timestamp(System.currentTimeMillis());
}

File: src/main/kotlin/com/kevin/data/dal/entity/Counter.kt
package com.kevin.data.dal.entity

import java.sql.Timestamp
import javax.persistence.*

@Entity
@Table(name = "counter")
class Counter(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int=0,

    @Column(nullable = false)
    var count: Int=0,

    @Column(nullable = false, name = "create_time")
    val createTime: Timestamp = Timestamp(System.currentTimeMillis())
)
